add token param to authorize

CalvinChanCan opened the pull request:
This PR may be a bit more opinionated so feel free to ignore. For my use case, I'd like to be able to provide an auth token as a string rather than using the pickled session file. Personally, I think the developer should pickle the session file outside of this project. But to maintain that compatibility while allowing for using an auth token string, this PR does the following:

-  Sets the auth token in the headers while instantiating. This allows the client to authorize without needing to call login() (assuming the token is valid).
-  Remove cookies variables. The auth token is the only requirement to authorize the client.
-  Fixes the issue when saving the session file if the session directory does not exist by checking and creating the session directory.

I appreciate any feedback. Thanks!

hammem commented on monarchmoney/monarchmoney.py:
```diff
@@ -38,13 +39,17 @@ class LoginFailedException(Exception):
-    def __init__(self, session_file: str = SESSION_FILE, timeout: int = 10) -> None:
-        self._cookies = None
+    def __init__(
+        self, session_file: str = SESSION_FILE, timeout: int = 10, token: str = None
```
`token: Optional[str] = None`

hammem commented on monarchmoney/monarchmoney.py:
```diff
@@ -14,11 +14,12 @@
 ERRORS_KEY = "error_code"
-SESSION_FILE = ".mm/mm_session.pickle"
+SESSION_DIR = ".mm"
+SESSION_FILE = f"{SESSION_DIR}/mm_session.pickle"
```
good call!

hammem reviewed (commented):
Thanks for putting this up!  I get the use-case here and it makes sense.  Wondering if, instead of changing how the existing `*_session()` stuff works, we add a `token` property to this class and put logic in the `set_token()` method to handle setup.

We'd still allow for passing the token in the constructor, to make instantiation clean/easy for those opting not to have the class persist the entirety of the session.

Put in API terms:

  * Add `token` to the constructor
  * Add a `token` property and corresponding ``set_token()`
  * Refactor the class to use the new property for token handling
  * Leave the `*_session()` stuff as is, aside from using the new property internally

This should provide the flexibility you seek, which makes sense, while also providing a place to handle persistence of other things down the road (e.g. caching).

On the removal of the cookie jar from the persistence, oddly, I think Monarch *used to* have a cookie for it to recognize the device and avoid re-prompting for an MFA token.  But, looking at it lately, there's nothing in the jar!  Torn on removing it, in case they decide to change things...

CalvinChanCan commented:
Thanks @hammem for the feedback. 

I've made the suggested changes. As for avoiding the re-prompting for an MFA token, I've added a way to generate the TOTP if the MFA secret key is provided. Hopefully that would mean we wouldn't need cookies if we can now bypass the MFA prompt. I've added where to find the MFA secret in the README.
